core/arch/arm/kernel/secure_partition.c

```diff
@@ -170,14 +173,12 @@ bool sp_has_exclusive_access(struct sp_mem_map_region *mem,
 	return !sp_mem_is_shared(mem);
 }
 
-static uint16_t new_session_id(struct sp_sessions_head *open_sessions)
+static uint16_t new_session_id(void)
 {
-	struct sp_session *last = NULL;
 	uint16_t id = SPMC_ENDPOINT_ID + 1;
 
-	last = TAILQ_LAST(open_sessions, sp_sessions_head);
-	if (last)
-		id = last->endpoint_id + 1;
+	if (most_recently_opened)
+		id = most_recently_opened->endpoint_id + 1;
```

Comment:
I don't think works as it should (it was probably already broken). The function `read_manifest_endpoint_id()` handles conflicts with already loaded SPs, but `new_session_id()` doesn't do the same. What if `new_session_id()` tries to use an ID that has already been used by `read_manifest_endpoint_id()`?

Reply:
I made a correction to always search for the next available endpoint_id and use that to avoid duplication. If it is defined by a manifest of an SP to be loaded after, the "endpoint switching" mechanism implemented in read_manifest_endpoint_id will solve the collision.